modules/openapi-generator/src/main/resources/Java/RFC3339InstantDeserializer.mustache

```diff
@@ -0,0 +1,74 @@
+{{>licenseInfo}}
+package {{invokerPackage}};
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.Temporal;
+import java.time.temporal.TemporalAccessor;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer;
+
+{{>generatedAnnotation}}
+public class RFC3339InstantDeserializer<T extends Temporal> extends InstantDeserializer<T> {
+
+    public static final RFC3339InstantDeserializer<Instant> INSTANT = new RFC3339InstantDeserializer<>(
+        Instant.class, DateTimeFormatter.ISO_INSTANT,
+        Instant::from,
+        a -> Instant.ofEpochMilli( a.value ),
+        a -> Instant.ofEpochSecond( a.integer, a.fraction ),
+        null,
+        true // yes, replace zero offset with Z
+    );
+
+    public static final RFC3339InstantDeserializer<OffsetDateTime> OFFSET_DATE_TIME = new RFC3339InstantDeserializer<>(
+        OffsetDateTime.class, DateTimeFormatter.ISO_OFFSET_DATE_TIME,
+        OffsetDateTime::from,
+        a -> OffsetDateTime.ofInstant( Instant.ofEpochMilli( a.value ), a.zoneId ),
+        a -> OffsetDateTime.ofInstant( Instant.ofEpochSecond( a.integer, a.fraction ), a.zoneId ),
+        (d, z) -> ( d.isEqual( OffsetDateTime.MIN ) || d.isEqual( OffsetDateTime.MAX ) ?
+        d :
+        d.withOffsetSameInstant( z.getRules().getOffset( d.toLocalDateTime() ) ) ),
+        true // yes, replace zero offset with Z
+    );
+
+    public static final RFC3339InstantDeserializer<ZonedDateTime> ZONED_DATE_TIME = new RFC3339InstantDeserializer<>(
+        ZonedDateTime.class, DateTimeFormatter.ISO_ZONED_DATE_TIME,
+        ZonedDateTime::from,
+        a -> ZonedDateTime.ofInstant( Instant.ofEpochMilli( a.value ), a.zoneId ),
+        a -> ZonedDateTime.ofInstant( Instant.ofEpochSecond( a.integer, a.fraction ), a.zoneId ),
+        ZonedDateTime::withZoneSameInstant,
+        false // keep zero offset and Z separate since zones explicitly supported
+    );
+
+    protected RFC3339InstantDeserializer(
+            Class<T> supportedType,
+            DateTimeFormatter formatter,
+            Function<TemporalAccessor, T> parsedToValue,
+            Function<FromIntegerArguments, T> fromMilliseconds,
+            Function<FromDecimalArguments, T> fromNanoseconds,
+            BiFunction<T, ZoneId, T> adjust,
+            boolean replaceZeroOffsetAsZ) {
+        super(
```

Comment:
This super constructor is deprecated (see [here](https://www.javadoc.io/doc/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.17.1/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.html#InstantDeserializer-java.lang.Class-java.time.format.DateTimeFormatter-java.util.function.Function-java.util.function.Function-java.util.function.Function-java.util.function.BiFunction-boolean-)). Could you please switch to a different one? I guess you could simply use the one recommended in the above link. And for the additional parameters required there, you could use the default values. (DEFAULT_NORMALIZE_ZONE_ID=true, DEFAULT_ALWAYS_ALLOW_STRINGIFIED_DATE_TIMESTAMPS=false)

Reply:
@beikov
I see you marked this comment as resolved. But the code isn't changed. Maybe you have the changes locally and still need to push them?

Reply:
Yes, just pushed now